python/ray/data/preprocessor.py

```diff
@@ -58,8 +58,16 @@ class FitStatus(str, Enum):
         PARTIALLY_FITTED = "PARTIALLY_FITTED"
         FITTED = "FITTED"
 
+    class TransformStatus(str, Enum):
+        """The transform status of preprocessor."""
+
+        NOT_TRANSFORMED = "NOT_TRANSFORMED"
+        TRANSFORMED = "TRANSFORMED"
+
```

Comment:
My logic here is to prevent `inverse_transform` being called before `transform` is, regardless of the original dataset. So I suppose I'm not really recording dataset state, rather whether `transform` has been called by the preprocessor. 

Open to other alternatives on handling this though.